power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -435,7 +432,7 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
                                      ComponentFlags const& to_store) const {
         // TODO: (jguo) this function could be encapsulated in UpdateCompIndependence in update.hpp
         return run_functor_with_all_types_return_array([this, scenario_idx, &update_data, &to_store]<typename CT>() {
-            if (!to_store[index_of_component<CT>]) {
+            if (!std::get<index_of_component<CT>>(to_store)) {
```

Comment:
Nitpick: Rename `to_store` in this function to something more expressive. Maybe `available_components`, or `components_to_store`?